src/optvar.erl

```diff
@@ -137,8 +137,13 @@ read(Key, Timeout) ->
                 %% transmitted in a DOWN message from a temporary
                 %% "waker" process. See `waker_loop':
                 {'DOWN', MRef, _, _, Reason} ->
-                    {optvar_set, Value} = Reason, % Assert
-                    {ok, Value}
+                    case Reason of % Assert
+                        {optvar_set, Value} ->
+                            {ok, Value};
+                        noproc ->
+                            %% race condition; retry
+                            read(Key, Timeout)
```

Comment:
This can exceed the `Timeout` (since time spent waiting is not subtracted)

Reply:
:heavy_check_mark: